src/enumerable.cr

```diff
@@ -535,6 +537,15 @@ module Enumerable(T)
     ary
   end
 
+  private def flat_map_type(elem)
+    case elem
+    when Array, Iterator
+      elem.first
```

Comment:
Shouldn't this be `flat_map_type(elem.first)`? 🤔

Reply:
flat_map flattens 1 level only.

Reply:
Oh, right! 💡